packages/server/src/routes/tenants.js

```diff
@@ -1,27 +1,37 @@
 const express = require('express');
 
 const router = express.Router({ mergeParams: true });
-const { requireAdminUser, requireUser } = require('../lib/access-helpers');
+const { requireUSDRSuperAdminUser } = require('../lib/access-helpers');
 const {
-    getTenant, setTenantDisplayName,
+    getTenants, setTenantDisplayName, knex,
 } = require('../db');
+const { createTenant, validateCreateTenantOptions } = require('../db/tenant_creation');
 
-router.get('/', requireUser, async (req, res) => {
-    const { user } = req.session;
-    let response;
-    if (user.role.name === 'admin') {
-        response = await getTenant(req.session.selectedAgency);
-    } else {
-        throw new Error(`You dont have access to tenants`);
-    }
-    res.json(response);
+router.get('/', requireUSDRSuperAdminUser, async (req, res) => {
+    const allTenants = await getTenants();
+    res.json(allTenants);
 });
 
-router.put('/:tenant', requireAdminUser, async (req, res) => {
-    // Currently, tenants are seeded into db; only display name is mutable.
-    const { tenant } = req.params;
+router.post('/', requireUSDRSuperAdminUser, async (req, res) => {
+    const result = await knex.transaction(async (trns) => {
+        const options = req.body;
+        const valid = await validateCreateTenantOptions(options, trns);
+        if (valid !== true) {
+            res.status(400).json({ error: valid });
+            return null;
+        }
+
+        return createTenant(options, trns);
+    });
+
+    res.json(result);
+});
+
+router.put('/:tenantId', requireUSDRSuperAdminUser, async (req, res) => {
+    // Only display name is mutable after creation.
+    const { tenantId } = req.params;
     const { displayName } = req.body;
-    const result = await setTenantDisplayName(tenant, displayName);
+    const result = await setTenantDisplayName(tenantId, displayName);
```

Comment:
Are these routes used anywhere yet?  I'd just as soon not support creation of a tenant via API if not.

Reply:
The "list tenants" and "update display name" do have UI that calls them; the "create tenant" one does not yet.

It's not pressing since we have the script implemented in #193, but this PR was open already and I wanted to make sure that if/when it did merge it correctly handled the ARPA tables.

Reply:
Filed follow-up task #358 to implement such UI.